File: Algorithms/Warmup/Plus Minus/plus_minus.py
def main():
    N = input()
    A = map(int, raw_input().split())

    zero, positive, negative = 0, 0, 0
    for a in A:
        if a > 0:
            positive += 1
        elif a < 0:
            negative += 1
        else:
            zero += 1

    total = float(len(A))

    print "%.3f\n%.3f\n%.3f\n" % ((positive / total), (negative / total), (zero / total))

if __name__ == '__main__':
    main()

File: Algorithms/Warmup/Plus Minus/plus_minus.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv)
{
    int N, *A, zero, positive, negative;

    scanf("%d", &N);
    A = malloc(sizeof(int) * N);

    zero = 0, positive = 0, negative = 0;
    for(int n = 0; n < N; ++n)
    {
        scanf("%d", &A[n]);

        if(A[n] == 0)
            ++zero;
        if(A[n] > 0)
            ++positive;
        if(A[n] < 0)
            ++negative;
    }

    printf("%.3f\n", (positive / (float) N));
    printf("%.3f\n", (negative / (float) N));
    printf("%.3f\n", (zero / (float) N));

    return 0;
}

